README.md

```diff
@@ -69,6 +69,27 @@ hCaptcha
         Log.d("hCaptcha", "hCaptcha is now visible.");
     }
   });
+  .addOnCloseListener(new OnCloseListener() {
+    @Override
+    public void onClose() {
+        // Usefull for analytics purposes
+        Log.d("hCaptcha", "hCaptcha is closed");
+    }
+  })
+  .addOnExpiredListener(new OnExpiredListener() {
+    @Override
+    public void onExpired() {
+        // Usefull for analytics purposes
+        Log.d("hCaptcha", "hCaptcha is expired");
+    }
+  })
+  .addOnChallengeExpiredListener(new OnChallengeExpiredListener() {
```

Comment:
I know my comment is a bit late but, instead of introducing new listeners, IMHO it would be better to follow the [execute async approach](https://docs.hcaptcha.com/configuration#asynchronous-mode) where we throw `onFailure` when challenge expired, or challenge was closed, etc. Here are [the error codes docs](https://docs.hcaptcha.com/configuration#error-codes).

The above approach would allow to keep the # of listeners small (as prior to this PR) and instead reuse the error listener for reporting close/expired/challenge-expired.

What do you reckon @e271828- @CAMOBAP  ?

Reply:
@DSergiu good proposal, to me it also seems better than adding a new callback API

Reply:
@e271828- if you agree with @DSergiu proposal too, I will update the PR

Reply:
@CAMOBAP go for it.
@DSergiu the app is responsible for tracking whether a token was consumed in all of our SDKs. I suppose we could add a `markUsed()` function or similar that simply cancels the expired callback, but not convinced it's necessary.

Reply:
Done, ready for review again

Reply:
I see that the `addOnChallengeExpiredListener` and `addOnCloseListener` were not removed. My suggestion above basically makes these obsolete since we report errors through `onFailure` for each of this events.

Reply:
Maybe we keep `SESSION_TIMEOUT` for challenge expiration and introduce `TOKEN_TIMEOUT` which is triggered after 2min (default) unless `markUsed` was called. @CAMOBAP

Reply:
Done TOKEN_TIMEOUT with id 16, once approved I will do the same for iOS